### ISSUE TITLE
scikit-learn BaseEstimator compliance 

### ISSUE THREAD

**Issue by ChristianSch** [scikit-multilearn/scikit-multilearn#24]
Hey there,

Background: methods like grid search and k-fold cross validation need an estimator to comply to the API stated [here](http://scikit-learn.org/stable/developers/contributing.html#rolling-your-own-estimator). The problem is that `MLClassifierBase` misses the `get_params` and `set_params` method that allows cloning of estimators as used by grid search and the `cross_validation` module (see [here](http://scikit-learn.org/stable/developers/contributing.html#cloning)).

The solution would either be to inherit `BaseEstimator` or to implement these mentioned methods. I patched them on my distribution because I use cross validation and depend on it. If you're up to it I'd provide a PR and you could review it if interested.

Cheers!

**Comment by niedakh** [comment]
Sure! Thank you, if you could provide a PR, I'd love to review it and put it in.

**Comment by ChristianSch** [comment]
Ok, awesome. I would provide a test (i.e. iris dataset) case for k-fold cross validation via scikit-learn classes that would both test this and #25.  What do you think? The notebooks seem to be deprecated.

**Comment by niedakh** [comment]
Hi @ChristianSch, 

I've applied your patch, thank you very much for checking this. I'm in the process of writing unit tests for all classifiers. One of which will include check_estimator.

I would love to see your use case of an iris k-fold, this could be a love addition to the documentation, if you provided the code for the example, I could write the docs for them, and we'd have our first use case tutorial!

**Comment by ChristianSch** [comment]
Awesome, thanks. Got fed up with my bsc thesis (2 days to go!) so I haven't gotten around merging your stuff into my branch to make a pull request. This got complicated the more I wated.

I'd love to contribute. This would simply things a lot! And I'm totally up for contributing the iris k-fold cross validation tut. I'll look into this in the next days. (Is there an IRC channel or something?)

PS: Good you reminded me, I replaced a link to GitHub with the bibtex from the website in my thesis. Would've missed that otherwise.
